### ISSUE TITLE
Implement element functor spatial argument for elemental variables

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#25187]
## Reason
The `ElemArg` functor spatial argument (https://mooseframework.inl.gov/syntax/Functors/index.html) is not implemented for variables, but it could at least be implemented for elemental variables. One place where this could be used is `ElementExtremeFunctorValue`.

## Design
This would implement the `ElemArg` functor spatial argument for elemental variables.

## Impact
New capability.

**Comment by joshuahansel** [comment]
@lindsayad for your interest. Not urgent, just something I ran into and will go around.

**Comment by joshuahansel** [comment]
I also wonder if we can (and should) make some implementation for "element" values of nodal variables, even outside of functors. The choices would be the numerical average value of the variable on the element and the value at the centroid.

**Comment by MengnanLi91** [comment]
I have time to work on this if no one has already started.

**Comment by joshuahansel** [comment]
> I have time to work on this if no one has already started.

I haven't started on it - it's nothing urgent (at least for me). I was just wanting to use it in a test, but I just ended up taking a different strategy.

I do think some kind of solution to what I encountered for `ElementExtremeFunctorValue` would be useful. Maybe it's this, or maybe we just need to enable `ElementExtremeFunctorValue` to work with `ElemQpArg` too 🤔 Thoughts @GiudGiud @lindsayad ?
